builtins/to-toml.nix

```diff
@@ -25,106 +25,115 @@ let
+            else abort "Not implemented: type ${ty}"
+  ;
 
   outputKeyValInner = k: v:
-    let ty = tomlTy v; in
-    if ty == "set" then
-      let
-        vals = mapAttrsToList
-          (k': v': "${quoteKey k'} = ${outputValInner v'}") v;
-        valsStr = concatStringsSep ", " vals;
-      in
-      [ "${quoteKey k} = { ${valsStr} }" ] else
-    outputKeyVal k v;
+    let
+      ty = tomlTy v;
+    in
+      if ty == "set" then
+        let
+          vals = mapAttrsToList
+            (k': v': "${quoteKey k'} = ${outputValInner v'}") v;
+          valsStr = concatStringsSep ", " vals;
+        in
+          [ "${quoteKey k} = { ${valsStr} }" ] else
+        outputKeyVal k v;
 
   # Returns a list of strings; one string per line
   outputKeyVal = k: v:
-    let ty = tomlTy v; in
-    if ty == "bool" then
-      [ "${quoteKey k} = ${outputValInner v}" ]
-      else
-    if ty == "string" then
-      [ "${quoteKey k} = ${quoteString v}" ]
-      else
-    if ty == "list_of_attrs" then
-      concatMap (inner:
-      [ "[[${k}]]" ] ++
-      (concatLists (mapAttrsToList outputKeyValInner inner))) v
+    let
+      ty = tomlTy v;
+    in
+      if ty == "bool" then
+        [ "${quoteKey k} = ${outputValInner v}" ]
       else
-    if ty == "list" then
-      let
-        vals = map quoteString v;
-        valsStr = concatStringsSep ", " vals;
-        in
-      [ "${quoteKey k} = [ ${valsStr} ]" ] else
-    if ty == "set" then
-      [ "[${k}]" ] ++
-      (concatLists (mapAttrsToList outputKeyValInner v))
-    else abort "Not implemented: type ${ty} for key ${k}"
-      ;
+        if ty == "string" then
+          [ "${quoteKey k} = ${quoteString v}" ]
+        else
+          if ty == "list_of_attrs" then
+            concatMap (
+              inner:
+                [ "[[${k}]]" ] ++ (concatLists (mapAttrsToList outputKeyValInner inner))
+            ) v
+          else
+            if ty == "list" then
+              let
+                vals = map quoteString v;
+                valsStr = concatStringsSep ", " vals;
+              in
+                [ "${quoteKey k} = [ ${valsStr} ]" ] else
+              if ty == "set" then
+                [ "[${k}]" ] ++ (concatLists (mapAttrsToList outputKeyValInner v))
+              else abort "Not implemented: type ${ty} for key ${k}"
+  ;
 
   tomlTy = x:
     if typeOf x == "string" then "string" else
-    if typeOf x == "bool" then "bool" else
-    if typeOf x == "int" then "int" else
-    if typeOf x == "float" then "float" else
-    if typeOf x == "set" then
-      if lib.isDerivation x then "string" else "set" else
-    if typeOf x == "list" then
-      if length x == 0 then "list"
-      else
-        let ty = typeOf (elemAt x 0); in
-        #assert (all (v: typeOf v == ty) x);
-        if ty == "set" then "list_of_attrs" else "list"
-    else abort "Not implemented: toml type for ${typeOf x}";
+      if typeOf x == "bool" then "bool" else
```

Comment:
mh this is a bit annoying, it gets much harder to read

Reply:
according to some of the formatting later on in this module, putting the `else` on the next line should fix it

Reply:
what do you think of the new format?